gem/optimise.py

```diff
@@ -187,6 +187,12 @@ def child(expression):
     if types <= {Literal, Zero, Failure}:
         return partial_indexed(ListTensor(expressions), (index,))
 
+    if types <= {ComponentTensor, Zero}:
```

Comment:
Yes, because we cannot remove `ComponentTensor` directly inside `Concatenate` nodes. Similarly, we cannot remove `ComponentTensor` nodes at the root of the expression.